lib/middleware/routes-output/routes-output.js

```diff
@@ -175,6 +175,10 @@ const generateOutput = async (req, res) => {
 
 // Convenience route to generate output for current user
 router.use('/:outputType/:outputId/:destination?/:filename', useAsync(async (req, res) => {
+  // Only allow unforwarded requests access
+  if (req.get('x-forwarded-host')) {
+    throw new Error(404)
```

Comment:
Perhaps throwing a forbidden is a bit more appropriate here?
```suggestion
    throw new Error(403)
```